### PR TITLE
Account user forms

### PR DESCRIPTION
**Pull request by qbob1**
This branch contains the forms used to submit new payments and addresses. As of now all forms emit an event to be caught by the appropriate parent component and processed by the server.

### CODE REVIEW

File: angular/src/app/modules/account/account-routing.module.ts [hunk 1]
```diff
@@ -18,8 +21,10 @@ const routes: Routes = [{ component: AccountComponent, path: '' }];
     PaymentComponent,
     ProfileComponent,
     AccountReviewComponent,
+    NewaddressformComponent,
+    NewPaymentFormComponent,
   ],
   exports: [RouterModule],
-  imports: [CommonModule, RouterModule.forChild(routes)],
+  imports: [CommonModule, RouterModule.forChild(routes), FormsModule, ReactiveFormsModule],
```
Comment (escalonn) [inline angular/src/app/modules/account/account-routing.module.ts:28]: it looks like you're not doing template-driven forms so pretty sure you don't need FormsModule

File: angular/src/app/modules/account/newaddressform/newaddressform.component.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import { Component, OnInit, Output, EventEmitter } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Address } from '../../../data/address.model';
+
+@Component({
+  selector: 'uic-newaddressform',
+  templateUrl: './newaddressform.component.html',
+  styleUrls: ['./newaddressform.component.scss'],
+})
+export class NewaddressformComponent implements OnInit {
```
Comment (escalonn) [inline angular/src/app/modules/account/newaddressform/newaddressform.component.ts:10]: naming conventions. should be `new-address-form.component.ts` and `NewAddressFormComponent`.
camelCase for properties below

File: angular/src/app/modules/account/newpaymentform/newpaymentform.component.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import { Component, OnInit, Output, EventEmitter } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Payment } from '../../../data/payment.model';
+
+@Component({
+  selector: 'uic-newpaymentform',
+  templateUrl: './newpaymentform.component.html',
+  styleUrls: ['./newpaymentform.component.scss'],
+})
+export class NewPaymentFormComponent implements OnInit {
```
Comment (escalonn) [inline angular/src/app/modules/account/newpaymentform/newpaymentform.component.ts:10]: should be `new-payment-form.component.ts`

File: angular/src/app/modules/account/newpaymentform/newpaymentform.component.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  showModal = false;
+
+  @Output() newPayment: EventEmitter<Payment> = new EventEmitter<Payment>();
+
+  PaymentForm = new FormGroup({
```
Comment (escalonn) [inline angular/src/app/modules/account/newpaymentform/newpaymentform.component.ts:15]: camelCase for properties

File: angular/src/app/modules/account/newpaymentform/newpaymentform.component.html [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+                    </span>
+                  </div>
+                  <label for="Securty Number">Security Number</label>
+                  <div class="level is-flex">
+                    <input type="number" id="secnumber" name="secnumber" class="input is-medium level-item"
+                      placeholder="ex: 111" formControlName="SecurityNumber" />
+                    <span class="level-item">
+                      <i *ngIf="SecurityNumber.invalid && (SecurityNumber.touched || SecurityNumber.dirty)"
+                        class="level-item fas fa-ban has-text-danger" aria-hidden="true" 
+                        title="Please enter valid CC Securty number: 111"></i>
```
Comment (escalonn) [inline angular/src/app/modules/account/newpaymentform/newpaymentform.component.html:42]: spelling

File: angular/src/app/modules/account/address/address.component.ts [hunk 6]
```diff
@@ -7,8 +7,13 @@ import { Address } from '../../../data/address.model';
...
   @Input() address: Address;
+  @Output() addressEdited = new EventEmitter();
 
   constructor() {}
 
   ngOnInit(): void {}
+
+  addAddress(newAddress: Address) {
+    console.log(newAddress);
+  }
```
Comment (escalonn) [inline angular/src/app/modules/account/address/address.component.ts:18]: shouldn't we ultimately send this data to the accountservice; is here a good reason that's not in this PR?

File: angular/src/app/modules/account/newpaymentform/newpaymentform.component.html (hunk 5 above)
Comment (TrueFlame42) [reply angular/src/app/modules/account/newpaymentform/newpaymentform.component.html]: how did you spot this single typo? amazing

File: angular/src/app/modules/account/newpaymentform/newpaymentform.component.ts (hunk 4 above)
Comment (jack-pflug) [reply angular/src/app/modules/account/newpaymentform/newpaymentform.component.ts]: addressed https://github.com/2006-jun15-net/rvtr-app-campsite/commit/3551d918d6e1af7f55891055bec0e0f948a7df0a

File: angular/src/app/modules/account/newpaymentform/newpaymentform.component.html (hunk 5 above)
Comment (jack-pflug) [reply angular/src/app/modules/account/newpaymentform/newpaymentform.component.html]: addressed https://github.com/2006-jun15-net/rvtr-app-campsite/commit/3551d918d6e1af7f55891055bec0e0f948a7df0a

File: angular/src/app/modules/account/newpaymentform/newpaymentform.component.ts (hunk 3 above)
Comment (TrueFlame42) [reply angular/src/app/modules/account/newpaymentform/newpaymentform.component.ts]: bd77cb1dd32017db4ecb8ab9a05f2dd747a7fee5  fixed

File: angular/src/app/modules/account/account-routing.module.ts (hunk 1 above)
Comment (TrueFlame42) [reply angular/src/app/modules/account/account-routing.module.ts]: removed the need for FormsModule c18df3c50fe10c7a7df773a8b88e4feb2926c071

File: angular/src/app/modules/account/newaddressform/newaddressform.component.ts (hunk 2 above)
Comment (TrueFlame42) [reply angular/src/app/modules/account/newaddressform/newaddressform.component.ts]: fixed bd77cb1dd32017db4ecb8ab9a05f2dd747a7fee5

File: angular/package-lock.json [hunk 7]
```diff
@@ -21288,7 +21316,11 @@
...
           "dev": true,
-          "optional": true
+          "optional": true,
+          "requires": {
+            "bindings": "^1.5.0",
+            "nan": "^2.12.1"
+          }
         },
         "is-absolute-url": {
           "version": "3.0.3",
```
Comment (qbob1) [inline angular/package-lock.json:21326]: Needed to regenerate this after removing an accidental dep

File: .gitignore [hunk 8]
```diff
@@ -11,3 +11,6 @@ node_modules/
 
 # sonarqube
 .scannerwork/
+
+.vs/
+.vscode/
```
Comment (escalonn) [inline .gitignore:16]: remove; and undelete .vscode/extensions.json

File: angular/package-lock.json (hunk 7 above)
Comment (escalonn) [reply angular/package-lock.json:21326]: doesn't seem necessary; if you empty out your node_modules, revert your package.lock, and then `npm install`, things should be working again

File: angular/src/app/modules/account/new-address-form/new-address-form.component.scss [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+  float: right;
+  color: grey;
+}
+
+.modal-background {
+  background: white;
+}
+
+/* Responsive layout - when the screen is less than 800px wide, make the two columns stack on top of each other instead of next to each other (and change the direction - make the "cart" column go on top) */
+@media (max-width: 800px) {
```
Comment (escalonn) [inline angular/src/app/modules/account/new-address-form/new-address-form.component.scss:90]: ideally we'd rely on bulma for this stuff

File: angular/src/app/modules/account/new-address-form/new-address-form.component.spec.ts [hunk 10]
```diff
@@ -0,0 +1,100 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { FormControl, FormsModule } from '@angular/forms';
+import { NewAddressFormComponent } from './new-address-form.component';
+import { Address } from '../../../data/address.model';
+describe('NewaddressformComponent', () => {
```
Comment (escalonn) [inline angular/src/app/modules/account/new-address-form/new-address-form.component.spec.ts:6]: describe string mismatch with actual type

File: angular/src/app/services/account/account.service.ts [hunk 11]
```diff
@@ -51,6 +52,24 @@ export class AccountService {
     return this.apiUrl$.pipe(concatMap((url) => this.http.post<boolean>(url, account)));
   }
 
+  /**
+   * Represents the _Account Service_ `post` method
+   *
+   * @param account Account
+   */
+  postAddress(address: Address): Observable<boolean> {
+    return this.apiUrl$.pipe(concatMap((url) => this.http.post<boolean>(url, address)));
```
Comment (escalonn) [inline angular/src/app/services/account/account.service.ts:61]: we can't post an address to the same endpoint we post an account and a payment

File: angular/package-lock.json (hunk 7 above)
Comment (jack-pflug) [reply angular/package-lock.json]: addressed https://github.com/RVTR/rvtr-app-campsite/commit/52c4362651251aa8df98f5d204967560a7087470

File: .gitignore (hunk 8 above)
Comment (jack-pflug) [reply .gitignore]: addressed https://github.com/RVTR/rvtr-app-campsite/commit/52c4362651251aa8df98f5d204967560a7087470

File: angular/src/app/modules/account/new-address-form/new-address-form.component.spec.ts (hunk 10 above)
Comment (jack-pflug) [reply angular/src/app/modules/account/new-address-form/new-address-form.component.spec.ts]: addressed https://github.com/RVTR/rvtr-app-campsite/commit/52c4362651251aa8df98f5d204967560a7087470

File: angular/src/app/services/account/account.service.ts (hunk 11 above)
Comment (TrueFlame42) [reply angular/src/app/services/account/account.service.ts]: fixed 9d1c1878ff0e2b237628b277538e8da71e43a6a4

File: .gitignore (hunk 8 above)
Comment (escalonn) [reply .gitignore]: not addressed. extensions.json is still deleted.

File: angular/src/app/services/account/account.service.ts [hunk 12]
```diff
@@ -4,7 +4,8 @@ import { Observable } from 'rxjs';
 import { concatMap, map } from 'rxjs/operators';
 import { ConfigService } from '../config/config.service';
 import { Account } from '../../data/account.model';
-
+import { Address } from '../../../app/data/address.model';
+import { Payment } from '../../../app/data/payment.model';
```
Comment (escalonn) [inline angular/src/app/services/account/account.service.ts:8]: newline after imports

File: angular/src/app/modules/account/new-address-form/new-address-form.component.spec.ts [hunk 13]
```diff
@@ -3,7 +3,8 @@ import { By } from '@angular/platform-browser';
 import { FormControl, FormsModule } from '@angular/forms';
 import { NewAddressFormComponent } from './new-address-form.component';
 import { Address } from '../../../data/address.model';
-describe('NewaddressformComponent', () => {
+
+describe('NewAddressformComponent', () => {
```
Comment (escalonn) [inline angular/src/app/modules/account/new-address-form/new-address-form.component.spec.ts:7]: previous review comment not resolved. string _still_ mismatched with type. if you can't get small things right then you won't be trusted with big things.

File: angular/src/app/modules/account/address/address.component.spec.ts [hunk 14]
```diff
@@ -1,9 +1,10 @@
-import { HttpClientTestingModule } from '@angular/common/http/testing';
 import { of } from 'rxjs';
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { AddressComponent } from './address.component';
+import { ACCOUNT_EDITING_SERVICE } from '../account-editing.token';
 import { Account } from '../../../data/account.model';
-import { AccountService } from '../../../services/account/account.service';
+
```
Comment (escalonn) [inline angular/src/app/modules/account/address/address.component.spec.ts:6]: random empty line, but w/e

File: angular/src/app/modules/account/payment/payment.component.spec.ts [hunk 15]
```diff
@@ -17,6 +17,7 @@ describe('PaymentComponent', () => {
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
```
Comment (escalonn) [inline angular/src/app/modules/account/payment/payment.component.spec.ts:20]: not good; if there's a need for httpclient in a component test then that means you're probably using the real service, and instead it should be handled with test doubles for the services.

but w/e

**Pull request merged**